### PR TITLE
Feat--Pendle-swapper

### PR DESCRIPTION
**Pull request by GuillaumeNervoXS**

### CODE REVIEW

File: contracts/swapper/LevSwapper/pendle/PendleLevSwapper.sol [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    // =============================== MAIN FUNCTIONS ==============================
+
+    /// @inheritdoc BaseLevSwapper
+    function _add(bytes memory) internal override returns (uint256 amountOut) {
+        uint256 amount = collateral().balanceOf(address(this));
+        // It needs to be deposited directly onto `YT`contracts to mint both PT and YT tokens
+        uint256 amountSharesOut = SY().deposit(address(this), address(collateral()), amount, 0);
+        LimitOrderData memory limit;
```
Comment (sogipec) [inline contracts/swapper/LevSwapper/pendle/PendleLevSwapper.sol:35]: seems useless here

Comment (GuillaumeNervoXS) [reply contracts/swapper/LevSwapper/pendle/PendleLevSwapper.sol:35]: I have to create the empty object

File: contracts/swapper/LevSwapper/morpho/PendleLevSwapperTemplateMorpho.sol [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    constructor(
+        ICoreBorrow _core,
+        IUniswapV3Router _uniV3Router,
+        address _aggregator,
+        IAngleRouterSidechain _angleRouter,
+        IMorphoBase _morpho
+    ) PendleLevSwapperMorpho(_core, _uniV3Router, _aggregator, _angleRouter, _morpho) {}
+
+    /// @inheritdoc BaseLevSwapper
+    function angleStaker() public pure override returns (IBorrowStaker) {
```
Comment (sogipec) [inline contracts/swapper/LevSwapper/morpho/PendleLevSwapperTemplateMorpho.sol:18]: Can put the empty angleStaker() address directly at the level of the PendleLevSwapperMorpho I guess

**Review by 0xtekgrinder — COMMENTED** [review]
My only concern is why do we still wants to separate the transfer to the swapper into two txs ? If we add some argument to pull the funds it could be safer

File: contracts/swapper/LevSwapper/ERC4626LevSwapper.sol [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        uint256 amountOut = token().redeem(amount, address(this), address(this));
+        // We let this check because the caller may swap part of the tokens received and therefore
+        // the check in the base swapper will only be for the out token
+        if (amountOut < minAmountOut) revert TooSmallAmountOut();
+    }
+
+    // ============================= VIRTUAL FUNCTIONS =============================
+
+    /// @notice Reference to the underlying asset
+    function asset() public pure virtual returns (IERC20);
```
Comment (sogipec) [inline contracts/swapper/LevSwapper/ERC4626LevSwapper.sol:46]: maybe we can just encode token and asset in the bytes this way we just have one contract for everything

**Pull request merged**